recipes/protobuf/all/conanfile.py

```diff
@@ -41,36 +42,85 @@ def source(self):
     def configure(self):
         if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
             del self.options.fPIC
+
             compiler_version = Version(self.settings.compiler.version.value)
             if compiler_version < "14":
                 raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
                                                 "Visual Studio 2015 or higher.")
+            if self.options.shared and "MT" in self.settings.compiler.runtime:
+                raise ConanInvalidConfiguration("Protobuf can't be build with shared + MT(d) runtimes")
+
+        if tools.is_apple_os(self.settings.os) and self.options.shared:
+            raise ConanInvalidConfiguration("Protobuf could not be build as shared library for Mac.")
 
     def requirements(self):
-        if self.options.with_zlib:
+        if self.options.get_safe("with_zlib"):
             self.requires("zlib/1.2.11")
 
     @property
     def _cmake_install_base_path(self):
         return os.path.join("lib", "cmake", "protobuf")
 
     def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
-        self._cmake.definitions["protobuf_BUILD_TESTS"] = False
-        self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
-        self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = not self.options.lite
-        self._cmake.definitions["protobuf_BUILD_PROTOBUF_LITE"] = self.options.lite
```

Comment:
Is it guaranteed to always be enabled?

Reply:
I'm not entirely sure to which part you are referring. You mean the protoc binaries? We might want to set `protobuf_BUILD_PROTOC_BINARIES ` and `protobuf_BUILD_LIBPROTOC ` explicitly to True just to be safe

Reply:
I was talking about the options that are not set now, that were set previously.
Such as `protobuf_BUILD_PROTOBUF_LITE` and `protobuf_BUILD_PROTOC_BINARIES`.

Reply:
Done: https://github.com/conan-io/conan-center-index/pull/2037/commits/aaad499f68aee4aecd2d53f4270ed3f2a9cf2e36

`protobuf_BUILD_PROTOBUF_LITE` does not exist upstream. We added this with our previous patches